### PR TITLE
Bump wasmtime to 3.0.0

### PR DESCRIPTION
**Pull request by etehtsea**
Few fixes for compatibility with https://github.com/bytecodealliance/wasmtime/pull/5149

Signed-off-by: Konstantin Shabanov <mail@etehtsea.me>

### CODE REVIEW

File: tests/runtime/invalid/host.rs [hunk 1]
```diff
@@ -89,10 +88,10 @@ fn run(wasm: &str) -> Result<()> {
     )?;
     return Ok(());
 
-    fn assert_err(result: Result<(), Trap>, err: &str) -> Result<()> {
+    fn assert_err(result: anyhow::Result<()>, err: &str) -> anyhow::Result<()> {
         match result {
             Ok(()) => anyhow::bail!("export didn't trap"),
-            Err(e) if e.to_string().contains(err) => Ok(()),
+            Err(e) if format!("{}", e.root_cause()).contains(err) => Ok(()),
```
Comment (dicej) [inline tests/runtime/invalid/host.rs]: Just to be a bit more concise:
```suggestion
            Err(e) if e.root_cause().to_string().contains(err) => Ok(()),
```

**Review by dicej — APPROVED** [review]
Thanks for doing this!

File: tests/runtime/invalid/host.rs (hunk 1 above)
Comment (etehtsea) [reply tests/runtime/invalid/host.rs]: Fixed, thanks!

**Review by lann — COMMENTED** [review]
Thanks! Could you add a description of the error handling changes to the commit message? When maintaining a fork of an upstream project it is especially important to carefully track updates like that.

**Comment by etehtsea** [comment]
> Thanks! Could you add a description of the error handling changes to the commit message? When maintaining a fork of an upstream project it is especially important to carefully track updates like that.

done! Thanks!

**Comment by lann** [comment]
Nice commit message, thanks!

**Pull request merged**